pkg/monitoring/vms/prometheus/prometheus.go

```diff
@@ -161,34 +152,91 @@ func tryToPushMetric(desc *prometheus.Desc, mv prometheus.Metric, err error, ch
 }
 
 func updateMemory(vmi *k6tv1.VirtualMachineInstance, vmStats *stats.DomainStats, ch chan<- prometheus.Metric) {
-	if vmStats.Memory.AvailableSet {
-		mv, err := prometheus.NewConstMetric(
-			memoryAvailableDesc, prometheus.GaugeValue,
-			// the libvirt value is in KiB
-			float64(vmStats.Memory.Available)*1024,
-			vmi.Status.NodeName, vmi.Namespace, vmi.Name,
-			vmStats.Name,
-		)
-		tryToPushMetric(memoryAvailableDesc, mv, err, ch)
+	memoryResidentLabels = []string{"node", "namespace", "name"}
+	memoryAvailableLabels = []string{"node", "namespace", "name"}
+	swapTrafficLabels = []string{"node", "namespace", "name", "type"}
+	var memoryResidentLabelValues = []string{vmi.Status.NodeName, vmi.Namespace, vmi.Name}
+	var memoryAvailableLabelsValues = []string{vmi.Status.NodeName, vmi.Namespace, vmi.Name}
+	var swapTrafficInLabelsValues = []string{vmi.Status.NodeName, vmi.Namespace, vmi.Name, "in"}
+	var swapTrafficOutLabelsValues = []string{vmi.Status.NodeName, vmi.Namespace, vmi.Name, "out"}
+
+	// Add k8s metadata.Labels as metric labels
+	labelPreffix := "vmi_k8s_label_"
```

Comment:
Does this follow the pattern which we see on pod metrics in k8s which are exposed?

Reply:
Particularly, I've never seen this use-case anywhere before... Adding pod metadata to be exposed by another one.

So I can't say what is the best practice here

Reply:
Did you check the kubelet metrics?

Reply:
Are they supposed to be like this example?

![image](https://user-images.githubusercontent.com/24193764/85562841-3e069f80-b603-11ea-8201-c0131584c5a9.png)

`__meta_kubernetes_vmi_annotation_...` for annotations and `__meta_kubernetes_vmi_label_...` for labels

Reply:
or maybe 
`__meta_kubevirt_vmi`

Reply:
After some failed implementation and some research, I've noticed that it's not possible to add `__meta` labels to Prometheus metrics.

These sorts of labels are [reserved to Prometheus service-discoveries](https://github.com/prometheus/prometheus/tree/master/discovery#mapping-from-sd-to-prometheus), therefore I think the best approach would be `kubernetes_vmi_label_...` and `kubernetes_vmi_annotation...`

Maybe in the future, we could propose a Kubevirt service-discovery to Prometheus